docs/introduction-to-smart-contracts.rst

```diff
@@ -71,30 +71,23 @@ so that only you can alter the number.
 Subcurrency Example
 ===================
 
-The following contract will implement the simplest form of a
+The following contract implements the simplest form of a
 cryptocurrency. It is possible to generate coins out of thin air, but
-only the person that created the contract will be able to do that (it is easy
+only the person that created the contract is able to do that (it is possible
 to implement a different issuance scheme).
 Furthermore, anyone can send coins to each other without a need for
-registering with username and password — all you need is an Ethereum keypair.
-
+registering with username and password, all you need is an Ethereum keypair.
 
 ::
 
     pragma solidity >=0.5.0 <0.7.0;
 
     contract Coin {
-        // The keyword "public" makes those variables
```

Comment:
Don't you think these inline comments make it a bit more readable? With inline comments, you do not always have to flick back and forth and it is always clear (at least almost) what the text refers to.

Reply:
Personally I found it confusing that you see everything (without knowing what any of it is) and then you get the explanation, which for a beginner is a little confusing. Personally I had to go back and forth a lot with the current version to understand what lines meant, and re-read several times.

I have two ideas.

1. Add back the comments, but keep some of the further explanations after the code, offering people two ways to understand what the code does.
2. Change the order. Start with a blank contract, and add and explain each section piece by piece and then present the final code at the end. This is a more traditional way of structuring a getting started guide.

Reply:
I agree with you, but I think removing the comments exactly causes the problem that you see something without explanation ;)

Having said that, the exact comments might perhaps not be too helpful. Reading about "light clients" or things being "easily accessible from outside" does not really help people understand what this is about. We already know what state variables are, so this is good, but perhaps at least the comment on the constructor should stay? Perhaps you could also find better comments that explain in a single sentence what we are seeing here?